lib/validator/v3/validator.go

```diff
@@ -66,6 +67,13 @@ var (
 	// Hostname  have to be valid ipv4, ipv6 or strings up to 64 characters.
 	prometheusHostRegexp = regexp.MustCompile(`^[a-zA-Z0-9:._+-]{1,64}$`)
 
+	// global() cannot be used with other selectors.
+	andOr               = `(&&|\|\|)`
+	globalSelectorRegex = regexp.MustCompile(fmt.Sprintf(`%v global\(\)|global\(\) %v`, andOr, andOr))
+
+	// projectcalico.org/name cannot be used with other selectors.
+	namespaceSelectorRegex = regexp.MustCompile(fmt.Sprintf(`%v projectcalico.org/name == ".*"|projectcalico.org/name == ".*" %v`, andOr, andOr))
```

Comment:
This looks plausible to me, but:
- is it right to add this validation into the current PR?  is there a particular motivation for that?
- what about, for example, `projectcalico/name in {"red", "blue", ...}` - which I think should be allowed on its own, but probably not combined with anything else.

Unless there's a strong reason to include, I think it might be better to split the `projectcalico.org/name` validation out into another PR, so we can discuss its motivation and details more there.

Reply:
I think you're right - I'll remove the `projectcalico.org/name` bits